frontend/src/pages/Product.jsx

```diff
@@ -0,0 +1,109 @@
+import { useParams, Link, useNavigate } from 'react-router-dom';
+import { useState, useEffect } from 'react';
+import { LoremIpsum } from 'lorem-ipsum';
+import Header from '../components/Header';
+import ProductReviews from '../components/ProductReviews';
+import ConfirmationModal from '../components/ConfirmationModal';
+import { useGetProductByIdQuery, useDeleteProductByIdMutation } from '../services/products';
+
+const Product = () => {
+  const { productId } = useParams();
+  const { data, error, isLoading } = useGetProductByIdQuery(productId);
```

Comment:
It's really nice to be able to simply use this hook instead of writing all of this logic yourself.